### ISSUE TITLE
Center point to move entire polygons like rectangles

### ISSUE THREAD

**Issue by Zverik** [Leaflet/Leaflet.draw#243]
There is a point for rectangles, why not place the same marker for polygons? Finding its center is more tricky, of course, but can be done.

**Comment by w8r** [comment]
I would propose a slightly different approach to this. Take a look at my [Leaflet.draw.drag](https://github.com/w8r/Leaflet.draw.drag) and let me explain:
* It uses SVG/VML matrix transforms while dragging, so no recalculations on `mousemove` involved. This is crucial if you have to deal with the polygons that consist of hundreds of points. Reprojection is done only once - on release.
* It extends `L.Path` with `_applyTransform(matrix)` method, that gives way to all kinds of transformations that can be added to Lefalet.draw in the future: scale, skew, etc

@Zverik, @willfarrell, @jacobtoye, please review this plug-in, if you think it's useful, I can convert it into a pull-request

**Comment by dremonkey** [comment]
@w8r thanks for the plugin. From my testing so far, works like a charm. +1 to bringing this into Leaflet.Draw and also refactoring Edit.Rectangle and Edit.Circle to work like this, rather than needing the drag marker.

**Comment by w8r** [comment]
Added optional support for centroid markers, next thing I have in mind is turning that into PR

**Comment by willfarrell** [comment]
Finally had some time to take a quick look. Looks promising. I won't be able to take a closer look till I revisit my drawing code, which may be a while. Performance has been quite good in this area, but I expect it to be once users become more confident with using the tool and start creating very complex polygons.

**Comment by raoulalwani** [comment]
+1 to @w8r , thanks for the nice plugin. Leaving a comment so I am on the email notifications for this

**Comment by sharazam** [comment]
alright, did anything ever happen to this?

**Comment by willfarrell** [comment]
Nope, There is an open PR for it though. https://github.com/Leaflet/Leaflet.draw/pull/294
I'm no longer at the company I built this for. Perhaps someone else would like to take the lead on getting this merged in? I feel it's still a useful edit feature.
